src/services/refactors/inlineVariable.ts

```diff
@@ -191,7 +191,7 @@ function getReferenceNodes(declaration: InitializedVariableDeclaration, checker:
     const cannotInline = FindAllReferences.Core.eachSymbolReferenceInFile(declaration.name as Identifier, checker, file, ref => {
         // Only inline if all references are reads. Else we might end up with an invalid scenario like:
         // const y = x++ + 1 -> const y = 2++ + 1
-        if (FindAllReferences.isWriteAccessForReference(ref)) {
+        if (FindAllReferences.isWriteAccessForReference(ref) && !isShorthandPropertyAssignment(ref.parent)) {

```

Comment:
If changing this condition, consider updating the comment as well.

Reply:
Could you please explain why this change was needed? Does `FindAllReferences.isWriteAccessForReference` return true for `x`...

```ts
const x = 0;
const y = { x // ...here? }
```

Reply:
`ShorthandPropertyAssignment` is considered by `isWriteAccessForReference` as a write access reference.

https://github.com/microsoft/TypeScript/blob/6add96fd18d126d1505de5d96f0df6a86e5f2c21/src/services/findAllReferences.ts#L899-L902

https://github.com/microsoft/TypeScript/blob/6add96fd18d126d1505de5d96f0df6a86e5f2c21/src/services/findAllReferences.ts#L949